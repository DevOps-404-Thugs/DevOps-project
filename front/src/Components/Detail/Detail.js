import React from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import Modal from 'react-bootstrap/Modal';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Alert from 'react-bootstrap/Alert'

export class Description extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      address: "",
      prevState: null,
      needLogin: false,
      show: false,
      showSuccessBar: false,
      showFailureBar: false,
      errorCode: 0,
      goHomePage: false
    };
  }

  async componentDidMount() {
    const response = await fetch(`/housings/${this.props.objectId}`).then(res => res.json());
    this.setState(response);
  }

  handleInputChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleModify(e) {
    // Store current values to this.state.prevState
    this.setState({
      prevState: this.state,
      show: true
    })
  }

  async handleSave(e) {
    const response = await fetch(`/housings/${this.props.objectId}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      credentials: 'include',
      body: JSON.stringify(this.state)
    }).then(res => {
      if (res.status == 405) {
        this.setState({
          needLogin: true
        })
      } else if (res.status == 200) {
        this.setState({show: false, showSuccessBar: true});
      } else {
        this.state.prevState.showFailureBar = true;
        this.state.prevState.errorCode = res.status;
        this.setState(this.state.prevState);
      }
    });
  }

  async handleDelete(e) {
    const response = await fetch(`/housings/${this.props.objectId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      },
      credentials: 'include'
    }).then(res => {
      if (res.status == 405) {
        this.setState({
          needLogin: true
        });
      } else if (res.status == 200) {
        this.setState({
          goHomePage: true
        });
      } else {
        this.setState({
          show: false,
          showFailureBar: true,
          errorCode: res.status
        })
      }
    });
  }

  handleCancel(e) {
    // Restore the previous state
    this.state.prevState.show = false;
    this.setState(this.state.prevState);
  }

  handleClose(e) {
    this.setState({show: false});
  }

  handleCloseSuccessBar(e) {
    this.setState({showSuccessBar: false});
  }

  handleCloseFailureBar(e) {
    this.setState({showFailureBar: false});
  }

  render() {
    const {name, address, needLogin, show, showSuccessBar, 
      showFailureBar, errorCode, goHomePage} = this.state;
    if (needLogin) {
      return <Redirect to='/login' />
    }
    if (goHomePage) {
      return <Redirect to='/' />
    }
    const picNum = Math.abs(this.props.objectId.split("").reduce(function(a,b){a=((a<<5)-a)+b.charCodeAt(0);return a&a},0)) % 23;
    return (
      <>
        <Alert show={showSuccessBar} variant="success" 
            onClose={this.handleCloseSuccessBar.bind(this)} dismissible>
          Modified information successfully!
        </Alert>
        <Alert show={showFailureBar} variant="danger" 
            onClose={this.handleCloseFailureBar.bind(this)} dismissible>
          Failed to modify information, error response code: {errorCode}
        </Alert>
        <Modal show={show} onHide={this.handleClose.bind(this)}>
          <Modal.Header closeButton>
            <Modal.Title>Modify house information</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <InputGroup className="mb-3">
              <InputGroup.Prepend>
                <InputGroup.Text id="basic-addon3">Name</InputGroup.Text>
              </InputGroup.Prepend>
              <FormControl name="name" aria-describedby="basic-addon3" value={name} onChange={this.handleInputChange.bind(this)}/>
            </InputGroup>
            <InputGroup className="mb-3">
              <InputGroup.Prepend>
                <InputGroup.Text id="basic-addon3">Address</InputGroup.Text>
              </InputGroup.Prepend>
              <FormControl name="address" aria-describedby="basic-addon3" value={address} onChange={this.handleInputChange.bind(this)}/>
            </InputGroup>
          </Modal.Body>
          <Modal.Footer>
            <Button variant="primary" onClick={this.handleSave.bind(this)}>
              Save Changes
            </Button>
            <Button variant="secondary" onClick={this.handleCancel.bind(this)}>
              Cancel
            </Button>
          </Modal.Footer>
        </Modal>

        <Container>
          <Row>
            <Col></Col>
            <Col>
              <Card style={{ width: '50em', margin: '1em' }}>
                <Card.Img variant="top" src={process.env.PUBLIC_URL + "/pics/" + picNum + ".jpg"} />
                <Card.Body>
                  <Card.Title>{name}</Card.Title>
                  <Card.Subtitle className="mb-2 text-muted">{address}</Card.Subtitle>
                  <Card.Text>
                    A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers.
                  </Card.Text>
                  <Button variant="primary" onClick={this.handleModify.bind(this)}>Modify</Button>
                  <Button variant="primary" onClick={this.handleDelete.bind(this)} style={{ margin: '1px'}}>Delete</Button>
                </Card.Body>
              </Card>
            </Col>
            <Col></Col>
          </Row>
        </Container>
      </>
    );
  }
}

function Detail(props) {
  return (
    <Description objectId={props.location.state.objectId}/>
  );
}

export default withRouter(Detail);