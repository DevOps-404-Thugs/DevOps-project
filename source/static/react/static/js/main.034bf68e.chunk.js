(this.webpackJsonpfront=this.webpackJsonpfront||[]).push([[0],{108:function(e,t,a){},127:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),l=a(27),o=a.n(l),s=(a(90),a(17)),c=a(9),i=a(11),u=a(14),m=a.n(u),d=(a(108),a(46)),p=a(42),h=a(81),f=a(30);var g=function(){var e=Object(n.useState)(!1),t=Object(i.a)(e,2),a=t[0],l=t[1];return Object(n.useEffect)((function(){m.a.get("/login",{withCredentials:!0}).then((function(e){console.log("loginGET:"+e.status),console.log(e),200===e.status?l(!0):205===e.status&&l(!1)})).catch((function(e){console.log("loginGET:"+e.response.status),console.log(e.response)}))}),[]),console.log(a),r.a.createElement(d.a,{bg:"light",expand:"lg"},r.a.createElement(d.a.Brand,{href:"/ihomie"},"iHomie"),r.a.createElement(d.a.Toggle,{"aria-controls":"basic-navbar-nav"}),r.a.createElement(d.a.Collapse,{id:"basic-navbar-nav"},r.a.createElement(p.a,{className:"mr-auto"},r.a.createElement(p.a.Link,{href:"/ihomie"},"Home"),r.a.createElement(p.a.Link,{href:"/ihomie#/upload"},"Upload"),r.a.createElement(p.a.Link,{href:"/ihomie#/register"},"Register"),r.a.createElement(p.a.Link,{href:"/ihomie#/account"},"Account")),r.a.createElement(h.a,{inline:!0},!1===a?r.a.createElement(f.a,{variant:"outline-primary",onClick:function(e){window.location="/ihomie#/login"}},"Login"):r.a.createElement(f.a,{variant:"primary",onClick:function(e){m.a.get("/logout",{withCredentials:!0}).then((function(e){console.log("logoutGET:"+e.status),console.log(e),200===e.status&&l(!1)})).catch((function(e){console.log("logoutGET:"+e.response.status),console.log(e.response)}))}},"Logout"))))},E=a(132),b=a(131);var v=function(){var e=Object(n.useState)([]),t=Object(i.a)(e,2),a=t[0],l=t[1];Object(n.useEffect)((function(){m.a.get("/housings").then((function(e){console.log(e.data),console.log("housingsGET:"+e.status),l(e.data)}))}),[]);var o=a.map((function(e,t){var a=Math.abs(e._id.$oid.split("").reduce((function(e,t){return(e=(e<<5)-e+t.charCodeAt(0))&e}),0))%23;return r.a.createElement(E.a,{lg:6,md:8,xs:24,style:{border:"#CCC solid 1px","border-radius":"10px",padding:"30px",margin:"30px auto 30px"}},r.a.createElement("img",{style:{width:"100%"},src:"/static/react/pics/"+a+".jpg",alt:"houseImage"}),r.a.createElement("p",{style:{"padding-top":"10px",margin:"0","font-size":"24px","font-weight":"700"}},e.name),r.a.createElement("p",{style:{"padding-top":"10px","padding-bottom":"5px",margin:"0"}},e.address),r.a.createElement(s.b,{style:{"padding-top":"10px",margin:"0"},to:{pathname:"/detail",state:{objectId:e._id.$oid}}},"Go to see detail"))}));return r.a.createElement("div",{style:{width:"65%",margin:"3rem auto"}},r.a.createElement("div",{style:{textAlign:"center"}},r.a.createElement("h2",null,"  Find Your Future Home in iHomie ")),r.a.createElement("div",null,r.a.createElement(b.a,{gutter:[16,16]},o)),r.a.createElement("br",null),r.a.createElement("br",null),r.a.createElement("div",{style:{display:"flex",justifyContent:"center"}},r.a.createElement(s.b,{className:"link",to:"/upload"},"Upload New House")))},y=a(76),w=a(32),C=a.n(w),j=a(54),S=a(77),x=a(78),O=a(83),k=a(82),T=a(79),I=a(80),B=a(39),A=a(20),N=a(41),P=a(33),H=a(45),F=a(59),L=function(e){Object(O.a)(a,e);var t=Object(k.a)(a);function a(e){var n;return Object(S.a)(this,a),(n=t.call(this,e)).state={name:"",address:"",prevState:null,needLogin:!1,show:!1,showSuccessBar:!1,showFailureBar:!1,errorCode:0,goHomePage:!1},n}return Object(x.a)(a,[{key:"componentDidMount",value:function(){var e=Object(j.a)(C.a.mark((function e(){var t;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId)).then((function(e){return e.json()}));case 2:t=e.sent,this.setState(t);case 4:case"end":return e.stop()}}),e,this)})));return function(){return e.apply(this,arguments)}}()},{key:"handleInputChange",value:function(e){this.setState(Object(y.a)({},e.target.name,e.target.value))}},{key:"handleModify",value:function(e){this.setState({prevState:this.state,show:!0})}},{key:"handleSave",value:function(){var e=Object(j.a)(C.a.mark((function e(t){var a=this;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId),{method:"PUT",headers:{"Content-Type":"application/json"},credentials:"include",body:JSON.stringify(this.state)}).then((function(e){405==e.status?a.setState({needLogin:!0}):200==e.status?a.setState({show:!1,showSuccessBar:!0}):(a.state.prevState.showFailureBar=!0,a.state.prevState.errorCode=e.status,a.setState(a.state.prevState))}));case 2:e.sent;case 3:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"handleDelete",value:function(){var e=Object(j.a)(C.a.mark((function e(t){var a=this;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId),{method:"DELETE",headers:{"Content-Type":"application/json"},credentials:"include"}).then((function(e){405==e.status?a.setState({needLogin:!0}):200==e.status?a.setState({goHomePage:!0}):a.setState({show:!1,showFailureBar:!0,errorCode:e.status})}));case 2:e.sent;case 3:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"handleCancel",value:function(e){this.state.prevState.show=!1,this.setState(this.state.prevState)}},{key:"handleClose",value:function(e){this.setState({show:!1})}},{key:"handleCloseSuccessBar",value:function(e){this.setState({showSuccessBar:!1})}},{key:"handleCloseFailureBar",value:function(e){this.setState({showFailureBar:!1})}},{key:"render",value:function(){var e=this.state,t=e.name,a=e.address,n=e.needLogin,l=e.show,o=e.showSuccessBar,s=e.showFailureBar,i=e.errorCode,u=e.goHomePage;if(n)return r.a.createElement(c.a,{to:"/login"});if(u)return r.a.createElement(c.a,{to:"/"});var m=Math.abs(this.props.objectId.split("").reduce((function(e,t){return(e=(e<<5)-e+t.charCodeAt(0))&e}),0))%23;return r.a.createElement(r.a.Fragment,null,r.a.createElement(F.a,{show:o,variant:"success",onClose:this.handleCloseSuccessBar.bind(this),dismissible:!0},"Modified information successfully!"),r.a.createElement(F.a,{show:s,variant:"danger",onClose:this.handleCloseFailureBar.bind(this),dismissible:!0},"Failed to modify information, error response code: ",i),r.a.createElement(N.a,{show:l,onHide:this.handleClose.bind(this)},r.a.createElement(N.a.Header,{closeButton:!0},r.a.createElement(N.a.Title,null,"Modify house information")),r.a.createElement(N.a.Body,null,r.a.createElement(P.a,{className:"mb-3"},r.a.createElement(P.a.Prepend,null,r.a.createElement(P.a.Text,{id:"basic-addon3"},"Name")),r.a.createElement(H.a,{name:"name","aria-describedby":"basic-addon3",value:t,onChange:this.handleInputChange.bind(this)})),r.a.createElement(P.a,{className:"mb-3"},r.a.createElement(P.a.Prepend,null,r.a.createElement(P.a.Text,{id:"basic-addon3"},"Address")),r.a.createElement(H.a,{name:"address","aria-describedby":"basic-addon3",value:a,onChange:this.handleInputChange.bind(this)}))),r.a.createElement(N.a.Footer,null,r.a.createElement(f.a,{variant:"primary",onClick:this.handleSave.bind(this)},"Save Changes"),r.a.createElement(f.a,{variant:"secondary",onClick:this.handleCancel.bind(this)},"Cancel"))),r.a.createElement(T.a,null,r.a.createElement(I.a,null,r.a.createElement(B.a,null),r.a.createElement(B.a,null,r.a.createElement(A.a,{style:{width:"50em",margin:"1em"}},r.a.createElement(A.a.Img,{variant:"top",src:"/static/react/pics/"+m+".jpg"}),r.a.createElement(A.a.Body,null,r.a.createElement(A.a.Title,null,t),r.a.createElement(A.a.Subtitle,{className:"mb-2 text-muted"},a),r.a.createElement(A.a.Text,null,"A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers."),r.a.createElement(f.a,{variant:"primary",onClick:this.handleModify.bind(this)},"Modify"),r.a.createElement(f.a,{variant:"primary",onClick:this.handleDelete.bind(this),style:{margin:"1px"}},"Delete")))),r.a.createElement(B.a,null))))}}]),a}(r.a.Component);var D=Object(c.g)((function(e){return r.a.createElement(L,{objectId:e.location.state.objectId})}));var M=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),s=Object(i.a)(o,2),c=s[0],u=s[1];return r.a.createElement("div",null,r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"House Name"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Address"),r.a.createElement("input",{type:"text",onChange:function(e){u(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!c)return alert("Please fill in all fields!");console.log(a),console.log(c);var t={name:a,address:c};m.a.post("/housings",t,{withCredentials:!0}).then((function(e){console.log(e.status),console.log(e),200===e.status?alert("Upload successfully!"):alert("You meet with an error!")}))}},"Submit"))))};var U=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),c=Object(i.a)(o,2),u=c[0],d=c[1];return r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email address"),r.a.createElement("input",{type:"email",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Password"),r.a.createElement("input",{type:"password",onChange:function(e){d(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!u)return alert("Please fill in all fields!");var t={email:a,password:u};m.a.post("/login",t,{withCredentials:!0}).then((function(e){200===e.status?(alert("Login successfully!"),window.location.replace("/ihomie")):alert("You meet with an error!")})).catch((function(e){401===e.response.status?alert("You need to register for the account!"):400===e.response.status?alert("Password Error!"):402===e.response.status&&alert("Parameters Error!")}))}},"Submit")),r.a.createElement("small",{class:"text-muted"},"Need An Account? ",r.a.createElement(s.b,{className:"link",to:"/register"},"Sign Up Now")))};var Y=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),c=Object(i.a)(o,2),u=c[0],d=c[1],p=Object(n.useState)(""),h=Object(i.a)(p,2),f=h[0],g=h[1];return r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Username"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input username here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email address"),r.a.createElement("input",{type:"email",onChange:function(e){d(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Password"),r.a.createElement("input",{type:"password",onChange:function(e){g(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!u||!f)return alert("Please fill in all fields!");console.log(a),console.log(u),console.log(f);var t={username:a,email:u,password:f};console.log(t),m.a.post("/register",t).then((function(e){console.log(e.status),console.log(e),200===e.status?alert("Register successfully!"):alert("You meet with an error!")})).catch((function(e){console.log(e.response),401===e.response.status?alert("The email has been registered!"):402===e.response.status?alert("This username is unavailable!"):400===e.response.status?alert("You have loged in already!"):403===e.response.status&&alert("Parameter wrong!")}))}},"Submit")),r.a.createElement("small",{class:"text-muted"},"Already Have An Account? ",r.a.createElement(s.b,{className:"link",to:"/login"},"Sign In")))};var G=function(){var e=Object(n.useState)({username:null,password:null}),t=Object(i.a)(e,2),a=t[0],l=t[1];return Object(n.useEffect)((function(){m.a.get("/account").then((function(e){console.log(e.data),l(e.data)})).catch((function(e){console.log(e.response)})),console.log(a)})),r.a.createElement("div",{style:{width:"50%",margin:"3rem auto"}},null==a.username?r.a.createElement("h2",{style:{display:"flex",justifyContent:"center"}},"You need to login first!"):r.a.createElement("div",null,r.a.createElement(A.a,{style:{"margin-bottom":"30px"}},r.a.createElement(A.a.Header,{style:{textAlign:"center"}},"Account Info"),r.a.createElement(A.a.Title,{style:{"padding-top":"0.75rem","padding-left":"1rem"}},"username: ",a.username),r.a.createElement(A.a.Text,{style:{"padding-bottom":"0.75rem","padding-left":"1rem","font-weight":"bold"}},"email: ",a.email)),r.a.createElement(s.b,{className:"link",to:"/account_update",style:{color:"blue",display:"flex",justifyContent:"center"}},"Update Account Information")),r.a.createElement("br",null),r.a.createElement("br",null),r.a.createElement("div",{style:{display:"flex",justifyContent:"center"}}))};var _=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),s=Object(i.a)(o,2),c=s[0],u=s[1];return r.a.createElement("div",null,r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"User Name"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"username-input",placeholder:"Input new username here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email"),r.a.createElement("input",{type:"text",onChange:function(e){u(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input new email address here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!c)return alert("Please fill in all fields!");console.log(a),console.log(c);var t={username:a,email:c};m.a.put("/account",t,{withCredentials:!0}).then((function(e){console.log(e.status),console.log(e),200===e.status?(alert("Account updated successfully!"),window.location.replace("/ihomie")):alert("You meet with an error!")})).catch((function(e){console.log(e.response),400===e.response.status&&alert("This account has existed!")}))}},"Submit"))))};var J=function(){return r.a.createElement("div",{className:"App"},r.a.createElement(s.a,null,r.a.createElement(g,null),r.a.createElement(c.d,null,r.a.createElement(c.b,{exact:!0,path:"/",component:v}),r.a.createElement(c.b,{exact:!0,path:"/detail",component:D}),r.a.createElement(c.b,{exact:!0,path:"/upload",component:M}),r.a.createElement(c.b,{exact:!0,path:"/login",component:U}),r.a.createElement(c.b,{exact:!0,path:"/register",component:Y}),r.a.createElement(c.b,{exact:!0,path:"/account",component:G}),r.a.createElement(c.b,{exact:!0,path:"/account_update",component:_}))))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));a(126);o.a.render(r.a.createElement(r.a.StrictMode,null,r.a.createElement(J,null)),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)}))},85:function(e,t,a){e.exports=a(127)},90:function(e,t,a){}},[[85,1,2]]]);
//# sourceMappingURL=main.034bf68e.chunk.js.map