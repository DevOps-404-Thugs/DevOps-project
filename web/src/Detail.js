import React from 'react';

function Preview(props) {
  return (
    <div className="preview col-md-6">
      <div className="preview-pic tab-content">
        <div className="tab-pane active" id="pic-1"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></div>
        <div className="tab-pane" id="pic-2"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></div>
        <div className="tab-pane" id="pic-3"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></div>
        <div className="tab-pane" id="pic-4"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></div>
        <div className="tab-pane" id="pic-5"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></div>
      </div>
      <ul className="preview-thumbnail nav nav-tabs">
        <li className="active"><a data-target="#pic-1" data-toggle="tab"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></a></li>
        <li><a data-target="#pic-2" data-toggle="tab"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></a></li>
        <li><a data-target="#pic-3" data-toggle="tab"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></a></li>
        <li><a data-target="#pic-4" data-toggle="tab"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></a></li>
        <li><a data-target="#pic-5" data-toggle="tab"><img src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg" /></a></li>
      </ul>
    </div>
  );
}

export function Rating(props) {
  return (
    <div className="rating">
      <div className="stars">
        {props.stars >= 1 ? <span className="fa fa-star checked" /> : <span className="fa fa-star" />}
        {props.stars >= 2 ? <span className="fa fa-star checked" /> : <span className="fa fa-star" />}
        {props.stars >= 3 ? <span className="fa fa-star checked" /> : <span className="fa fa-star" />}
        {props.stars >= 4 ? <span className="fa fa-star checked" /> : <span className="fa fa-star" />}
        {props.stars >= 5 ? <span className="fa fa-star checked" /> : <span className="fa fa-star" />}
      </div>
      {typeof props.numReviews !== "undefined" && <span className="review-no">{props.numReviews} reviews</span>}
    </div>
  );
}

function DeleteButton(props) {
  function handleClick(e) {
  }

  return (
    <button className="add-to-cart btn btn-default" type="button">Delete</button>
  );
}

function ModifyButton(props) {
  function handleClick(e) {
  }

  return (
    <button className="add-to-cart btn btn-default" type="button">Modify</button>
  );
}

class Description extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "fake name",
      address: "fake address"
    };
  }

  async componentDidMount() {
    const response = await fetch("/housings/1").then(res => res.json());
    this.setState(response);
  }

  render() {
    const {name, address} = this.state;
    return (
      <div className="details col-md-6">
        <>
          <h3 className="product-title">{name}</h3>
          <p className="product-description">{address}</p>
          <Rating stars="2" numReviews="99"/>
          <p className="product-description">A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers.</p>
          <h4 className="price">current price: <span>$1800</span></h4>
          <p className="vote"><strong>Property ID</strong> 15599</p>
          <p className="vote"><strong>Location</strong> US</p>
          <p className="vote"><strong>Property Type</strong> House</p>
          <p className="vote"><strong>Status</strong> Rent</p>
  
          <div className="action">
            <>
              <DeleteButton />
              <ModifyButton />
            </>
          </div>
        </>
      </div>
    );
  }
}

/*
async function Description(props) {
  const response = await fetch("/housings/1").then(res => res.json());
  props.name = response.name;
  props.address = response.address;
  console.log(response);
  return (
    <div className="details col-md-6">
      <>
        <h3 className="product-title">{props.name}</h3>
        <p className="product-description">{props.address}</p>
        <Rating stars="2" numReviews="99"/>
        <p className="product-description">A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers.</p>
        <h4 className="price">current price: <span>$1800</span></h4>
        <p className="vote"><strong>Property ID</strong> 15599</p>
        <p className="vote"><strong>Location</strong> US</p>
        <p className="vote"><strong>Property Type</strong> House</p>
        <p className="vote"><strong>Status</strong> Rent</p>

        <div className="action">
          <>
            <DeleteButton />
            <ModifyButton />
          </>
        </div>
      </>
    </div>
  );
}
*/

function Container(props) {
  return (
    <>
    <div className="container">
      <div className="card">
        <div className="container-fliud">
          <div className="wrapper row">
            <>
              <Preview />
              <Description />
            </>
          </div>
        </div>
      </div>
    </div>
    </>
  );
}

function Detail(props) {
  return (
    <Container />
  );
}

export default Detail;