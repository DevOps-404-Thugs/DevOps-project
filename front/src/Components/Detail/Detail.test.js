import React from 'react';
import renderer from 'react-test-renderer';
import {Description} from './Detail.js';

test('Description generates correct page', () => {
    const component = renderer.create(
        <Description objectId="123" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

