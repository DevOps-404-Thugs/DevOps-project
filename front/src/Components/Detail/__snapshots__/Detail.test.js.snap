// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description generates correct page 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    />
    <div
      className="col"
    >
      <div
        className="card"
        style={
          Object {
            "margin": "1em",
            "width": "50em",
          }
        }
      >
        <img
          className="card-img-top"
          src="https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            
          </div>
          <div
            className="mb-2 text-muted card-subtitle h6"
          >
            
          </div>
          <p
            className="card-text"
          >
            A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers.
          </p>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Modify
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "margin": "1px",
              }
            }
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
    />
  </div>
</div>
`;
